src/wirecloud/catalogue/models.py

```diff
@@ -21,19 +22,28 @@
 from datetime import datetime
 from typing import Optional
 
+from src.settings import cache
 from src.wirecloud.commons.utils.template.schemas.macdschemas import MACD
 from src.wirecloud.database import Id
 
 
-class DBXhtml(BaseModel):
+class XHTML(BaseModel):
     uri: str
-    code: str
-    code_timestamp: Optional[datetime]
+    code: Optional[str] = None
+    code_timestamp: Optional[datetime] = None
     url: str
     content_type: Optional[str]
     use_platform_style: bool
     cacheable: bool
 
+    async def get_cache_key(self, resource_id: str, domain: str, mode: str, theme): #TODO: add theme type
```

Comment:
Si no sabes el tipo de theme pon `str` pero intenta no dejarlo vacío.
```suggestion
    async def get_cache_key(self, resource_id: str, domain: str, mode: str, theme: str): #TODO: add theme type
```